wandb/sdk/launch/builder/build.py

```
@@ -334,20 +336,59 @@ def get_requirements_section(launch_project: LaunchProject, builder_type: str) -
+            tomli = get_module("tomli")
+            if tomli is None:
+                wandb.termwarn(
+                    "pyproject.toml found but tomli could not be loaded. To "
+                    "install dependencies from pyproject.toml please run "
+                    "`pip install tomli` and try again."
+                )
+            else:
+                # First try to read deps from standard pyproject format.
+                with open(base_path / "pyproject.toml", "rb") as f:
+                    contents = tomli.load(f)
+                project_deps = [
+                    str(d) for d in contents.get("project", {}).get("dependencies", [])
+                ]
+                if project_deps:
+                    with open(base_path / "requirements.txt", "w") as f:
+                        f.write("\n".join(project_deps))
+                    requirements_files += ["src/requirements.txt"]
+                    pip_install_line = "pip install -r requirements.txt"
+                # If not found, check to see if we can install with poetry.
+                else:
+                    if "poetry" in contents.get("tool"):
+                        requirements_files += ["src/pyproject.toml"]
+                        pip_install_line = (
+                            "pip install poetry && "
+                            "poetry config virtualenvs.create false --local && "
+                            "poetry install --no-root"
+                        )
+        # Else use frozen requirements from wandb run.
+        if not pip_install_line and (base_path / "requirements.frozen.txt").exists():
+            requirements_files += [
+                "src/requirements.frozen.txt",
+                "_wandb_bootstrap.py",
+            ]
             pip_install_line = (
                 _parse_existing_requirements(launch_project)
                 + "python _wandb_bootstrap.py"
             )
+
+        if not pip_install_line:
```

This feels likely to build a lot of junk images... would it be better to just raise an error here?